File: 2559. Count Vowel Strings in Ranges/main.py
from typing import List


class Solution:
    def vowelStrings(self, words: List[str], queries: List[List[int]]) -> List[int]:
        def isValid(word):
            if not words: return False
            vowels = ['a', 'e', 'i', 'o', 'u']
            return word[0] in vowels and word[-1] in vowels

        prefix = [0] * (len(words) + 1)
        for i, w in enumerate(words):
            prefix[i+1] = prefix[i] + (1 if isValid(w) else 0)
        
        return [(prefix[r + 1] - prefix[l]) for l, r in queries]



        

File: 2559. Count Vowel Strings in Ranges/main.go
package main

import "strings"

func vowelStrings(words []string, queries [][]int) []int {
	isValid := func(word string) int {
		vowels := "aeiou"
		isValid := strings.ContainsRune(vowels, rune(word[0])) && strings.ContainsRune(vowels, rune(word[len(word)-1]))
		if isValid {
			return 1
		}
		return 0
	}

	prefix := make([]int, len(words)+1)

	for i, w := range words {
		prefix[i+1] = prefix[i] + isValid(w)
	}

	res := make([]int, len(queries))
	for i, q := range queries {
		l, r := q[0], q[1]
		res[i] = prefix[r+1] - prefix[l]
	}

	return res
}
